specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2023-12-01-preview/SystemComponents.json

```diff
@@ -0,0 +1,217 @@
+          "ClusterSystemComponents"
+        ],
+        "description": "Gets Kubernetes Cluster SystemComponents.",
+        "operationId": "SystemComponents_Get",
+        "x-ms-examples": {
+          "Get System component": {
+            "$ref": "./examples/GetSystemComponent.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterRpParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterResourceNameParameter"
+          },
+          {
+            "$ref": "../../common/2022-03-01/parameters.json#/parameters/ClusterNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SystemComponent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Component": {
+      "description": "System components registered in the cluster and its versions.",
+      "type": "object",
+      "readOnly": false,
+      "properties": {
+        "extensionType": {
+          "type": "string",
+          "description": "Extension type",
+          "readOnly": false
+        },
+        "userSpecifiedVersion": {
+          "type": "string",
+          "readOnly": false,
+          "description": "User specified version for the extension type."
+        },
+        "currentInstalledVersion": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Current installed version for the extension type."
+        },
+        "installedMajorVersion": {
+          "type": "integer",
+          "readOnly": true,
+          "description": "Current installed Major version for the extension type."
+        },
+        "status": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Status of the each components.",
+          "enum": [
+            "transitioning",
+            "success",
+            "failure"
+          ],
+          "default": "transitioning"
+        }
+      }
+    },
+    "SystemComponent": {
+      "description": "The SystemComponent for the cluster object.",
```

Comment:
It may be more useful to give a "user" description of what a system component is. Since this is the top-level resource for it the best place is probably on this description.

Reply:
+1